internal/controller/imagerepository_controller.go

```diff
@@ -181,6 +185,25 @@ func (r *ImageRepositoryReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	// Provision image repository if it hasn't been done yet
 	if !controllerutil.ContainsFinalizer(imageRepository, ImageRepositoryFinalizer) {
 		setMetricsTime(repositoryIdForMetrics, reconcileStartTime)
+		if isComponentLinked(imageRepository) {
+			componentExists, requeueAfterSeconds, err := r.CheckComponentExistence(ctx, imageRepository)
+			// getting component failed
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			// wait for component to appear, requeue without error
+			if requeueAfterSeconds > 0 {
+				return ctrl.Result{RequeueAfter: time.Duration(requeueAfterSeconds) * time.Second}, nil
+			}
+			// component doesn't exist and we won't requeue anymore
+			// 2 cases
```

Comment:
nitpick: it could go after a comma in the previous line.

Reply:
ok